frontend/web/components/AuditLog.tsx

```diff
@@ -96,17 +103,47 @@ const AuditLog: FC<AuditLogType> = (props) => {
     })
     const colour = index === -1 ? 0 : index
     let link: ReactNode = null
-    if (
-      related_object_uuid &&
-      related_object_type === 'EF_VERSION' &&
-      environment
-    ) {
+    const date = moment(created_date)
+    const isVersionEvent =
+      related_object_uuid && related_object_type === 'EF_VERSION' && environment
+    const versionLimitDays = subscriptionMeta?.feature_history_visibility_days
+    const isOverLimit = isVersionEvent
+      ? isVersionOverLimit(versionLimitDays, created_date)
+      : false
+    const VersionButton = (
+      <Button disabled={isOverLimit} theme='text'>
+        View version
+      </Button>
+    )
+    if (isVersionEvent) {
       link = (
-        <Link
-          to={`/project/${project.id}/environment/${environment.api_key}/history/${related_object_uuid}/`}
+        <Tooltip
+          title={
+            <div className='d-flex gap-2'>
+              {isOverLimit ? (
+                VersionButton
+              ) : (
+                <Link
+                  to={`/project/${project.id}/environment/${environment.api_key}/history/${related_object_uuid}/`}
+                >
+                  {VersionButton}
+                </Link>
+              )}
+              <PlanBasedBanner
+                force={isOverLimit}
+                feature={'VERSIONING'}
+                theme={'badge'}
+              />
+            </div>
+          }
         >
-          <Button theme='text'>View version</Button>
-        </Link>
+          {isOverLimit
+            ? `<div>
+              Unlock your feature's entire history.<br/>Currently limited to${' '}
+              <strong>${versionLimitDays} days</strong>.
```

Comment:
The wording here could be interpreted as unlocking this specific feature's history and not history on all features - maybe could be reworded as:

> Upgrade to access more than 7 days of feature history.

This avoids using negative terms (lock, limit) in this CTA and is generic enough to work for potential plans that might have more generous, but not unlimited, limits.